### PR TITLE
Added AutoSuggest and Debounce to LocationFinder

### PR DESCRIPTION
**Pull request by robinpiets**
This PR removes the Suffix field and adds an AutoSuggest box below the HouseNumber field. The LocationNotFound box is separated to a new component and is now debounced. I also added some general improvements to the LocationFinder component.

Please make sure:
- [x] to add a label
- [x] to add one or more reviewers
- [x] you followed our checklist for [functional testing](https://github.com/Amsterdam/vergunningcheck/blob/master/TESTING.md) **IE11 works :-D**
- [ ] ~~you added the necessary documentation (either in the markdown files or inline)~~ **This will be added in a new PR**
- [ ] ~~you added the necessary automated tests~~ **This will be added in a new PR**

### CODE REVIEW

File: apps/client/src/components/Location/LocationFinder.js [hunk 1]
```diff
@@ -36,18 +38,19 @@ const LocationFinder = (props) => {
   /* There is an issue with `skip`, it's not working if variables are given
      in `options` to `useQuery`. See https://github.com/apollographql/react-apollo/issues/3367
      Workaround is not giving any variables if the query should be skipped. */
-  const skip = !houseNumberFull && !postalCode;
+  const skip = !houseNumberFull || !postalCode;
```
Comment (robinpiets) [inline apps/client/src/components/Location/LocationFinder.js]: This fixes the red "cannot find address Error"

File: apps/client/src/components/Location/LocationFinder.js [hunk 2]
```diff
@@ -62,16 +65,22 @@ const LocationFinder = (props) => {
...
 
   // Validate forms
   const validate = (name, value, required) => {
     if (touched[name]) {
-      if (required && (!value || value?.trim() === "")) {
+      if (required && (!value || value?.toString().trim() === "")) {
         return requiredFieldText;
       }
-      const trimmed = value?.trim();
+      const trimmed = value && value.toString().trim();
```
Comment (robinpiets) [inline apps/client/src/components/Location/LocationFinder.js]: This fixes errors with undefined and numeric values

File: apps/client/src/components/Location/LocationFinder.js [hunk 3]
```diff
@@ -83,10 +92,32 @@ const LocationFinder = (props) => {
   const houseNumberError = validate("houseNumber", houseNumber, true);
 
   const handleBlur = (e) => {
-    setTouched({ ...touched, [e.target.name]: true });
+    e.target.value && setTouched({ ...touched, [e.target.name]: true });
```
Comment (robinpiets) [inline apps/client/src/components/Location/LocationFinder.js:101]: This fixes the focus error: when entering the page and you un-focus the default input we used to get an error on that input "this field should be set". This is now a thing from the past.

**Review by svenjens — APPROVED** [review]

File: apps/client/src/components/Location/LocationFinder.test.js [hunk 4]
```diff
@@ -17,74 +17,70 @@ const mockedFunctions = { ...{ setAddress, setFocus, setErrorMessage } };
...
     return (
-      <Context topicMock={topic} addressMock={mockedAddress}>
-        <LocationFinder {...props} {...mockedFunctions} />
+      <Context topicMock={topic} addressMock={mockedAddress.address}>
+        <LocationFinder topic={topic} {...props} {...mockedFunctions} />
       </Context>
     );
   };
 
   // @TODO enable these test again
```
Comment (afjlambert) [inline apps/client/src/components/Location/LocationFinder.test.js:38]: You can remove this comment.

**Pull request merged**